firewood/src/db.rs

```diff
@@ -870,85 +818,27 @@ impl Db {
         })
     }
 
-    /// Get a handle that grants the access to any committed state of the entire DB,
-    /// with a given root hash. If the given root hash matches with more than one
-    /// revisions, we use the most recent one as the trie are the same.
+    /// Get a handle that grants the access to any committed state of the entire DB
+    /// that is already in memory, with a given root hash. If the given root hash
+    /// matches with more than one revisions, we use the most recent one as the
+    /// trie are the same.
     ///
     /// If no revision with matching root hash found, returns None.
     // #[measure([HitCount])]
     pub fn get_revision(&self, root_hash: &TrieHash) -> Option<DbRev<StoreRevShared>> {
-        let mut revisions = self.revisions.lock();
+        let revisions = self.revisions.lock();
         let inner_lock = self.inner.read();
 
         // Find the revision index with the given root hash.
-        let mut nback = revisions.root_hashes.iter().position(|r| r == root_hash);
-        let rlen = revisions.root_hashes.len();
-
-        #[allow(clippy::unwrap_used)]
-        if nback.is_none() && rlen < revisions.max_revisions {
-            let ashes = inner_lock
-                .disk_requester
-                .collect_ash(revisions.max_revisions)
-                .ok()
-                .unwrap();
-
-            #[allow(clippy::indexing_slicing)]
-            (nback = ashes
-                .iter()
-                .skip(rlen)
-                .map(|ash| {
-                    StoreRevShared::from_ash(
-                        Arc::new(ZeroStore::default()),
-                        #[allow(clippy::indexing_slicing)]
-                        &ash.0[&ROOT_HASH_STORE_ID].redo,
-                    )
-                })
-                .map(|root_hash_store| {
-                    root_hash_store
-                        .get_view(0, TRIE_HASH_LEN as u64)
-                        .expect("get view failed")
-                        .as_deref()
-                })
-                .map(|data| TrieHash(data[..TRIE_HASH_LEN].try_into().unwrap()))
-                .position(|trie_hash| &trie_hash == root_hash));
-        }
-
+        let nback = revisions.root_hashes.iter().position(|r| r == root_hash);
```

Comment:
Nit: Can this line and the one below be combined by adding `?` to the end of this line?